pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -186,6 +186,11 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 					logger.Infof("orchestration status config map result channel closed, will restart watch.")
 					w.Stop()
 					<-time.After(100 * time.Millisecond)
+					opts.ResourceVersion = ""
+					st, err := c.context.Clientset.CoreV1().ConfigMaps(c.Namespace).List(opts)
+					if err == nil {
+						opts.ResourceVersion = st.ResourceVersion
```

Comment:
In addition to getting a new ResourceVersion, we also need to process the results of the List. Otherwise, we could miss status updates since the last watch. Up above, we check the process the status of the List on lines 132-167. We could make that into a helper function, which is also called here.